contracts/DebtToken.sol

```diff
@@ -155,93 +173,86 @@ contract DebtToken is Ownable {
   }
 
   /**
   Update the interest of the contract
   */
   function updateInterest() public {
-    require( loanMature() );
+    require( isTermOver() );
     uint interest_coins;
     uint256 interest_cycle;
     (interest_coins,interest_cycle) = calculateInterestDue();
     assert(interest_coins > 0 && interest_cycle > 0);
-    totalInterestCycle =  totalInterestCycle.add(interest_cycle);
-    lastinterestCycle = lastinterestCycle.add( interest_cycle.mul( interestCycleLength.mul(dayLength) ) );
-    mint(debtOwner , interest_coins);
+    totalInterestCycles =  totalInterestCycles.add(interest_cycle);
+    lastInterestCycle = lastInterestCycle.add( interest_cycle.mul( interestCycleLength.mul(dayLength) ) );
+    mint(lender , interest_coins);
   }
 
   /**
   Make payment to inititate loan
   */
   function fundLoan() public payable{
-    require(isDebtOwner(msg.sender));
-    require(msg.value > 0); //Ensure input available
-
-    uint256 weiValue = getLoanValue(true);
-    require(msg.value == weiValue);
-    require( balances[msg.sender] == 0); //Avoid double payment
+    require(isLender());
+    require(msg.value == getLoanValue(true)); //Ensure input available
+    require(!isLoanFunded()); //Avoid double payment
 
     balances[owner] = balances[owner].sub(totalSupply);
     balances[msg.sender] = balances[msg.sender].add(totalSupply);
+
     loanActivation = now;  //store the time loan was activated
-    lastinterestCycle = now.add(dayLength.mul(loanTerm) ) ; //store the date interest matures
-    owner.transfer(msg.value);
+    lastInterestCycle = now.add(dayLength.mul(loanTerm) ) ; //store the date interest matures
     mintingFinished = false;                 //Enable minting
+    
+    owner.transfer(msg.value);
     Transfer(owner,msg.sender,totalSupply);//Allow funding be tracked
   }
 
   /**
   Make payment to refund loan
   */
   function refundLoan() onlyOwner public payable{
-    if(! interestStatusUpdated() )
+    if(! isInterestStatusUpdated() )
         updateInterest(); //Ensure Interest is updated
 
-    require(msg.value > 0);
     require(msg.value == getLoanValue(false));
+    require(isLoanFunded());
 
-    require(balances[debtOwner] > 0);
     finishMinting() ;//Prevent further Minting
 
-    balances[debtOwner] = balances[debtOwner].sub(totalSupply);
+    balances[lender] = balances[lender].sub(totalSupply);
     balances[owner] = balances[owner].add(totalSupply);
-    debtOwner.transfer(msg.value);
-    Transfer(debtOwner,owner,totalSupply);//Allow funding be tracked
+
+    lender.transfer(msg.value);
+    Transfer(lender,owner,totalSupply);//Allow funding be tracked
   }
 
   /**
   Fallback function
   */
   function() public payable{
     require(initialSupply > 0);//Stop the whole process if initialSupply not set
-    if(msg.sender == owner && balances[msg.sender] == 0)
+    if(isBorrower())
       refundLoan();
-    else if(isDebtOwner(msg.sender) && balances[msg.sender] == 0)
+    else if(isLender())
```

Comment:
fund state check needed ` && !isLoanFunded()`

Reply:
Both `fundLoan` and `refundLoan` has `require` statement for isLoanFunded so it was redundant here.

Reply:
That is right,
missed that